### PR TITLE
Features/#372 sections memorizing

### PR DESCRIPTION
**Pull request by dakochik**
Issue #372 is resolved

### CODE REVIEW

File: src/main/kotlin/com/jetbrains/snakecharm/inspections/quickfix/AddIgnoredElementQuickFix.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        startElement: PsiElement,
+        endElement: PsiElement
+    ) {
+        val virtualFile = startElement.containingFile.virtualFile
+        if (virtualFile != null) {
+            val text = startElement.text
+            val inspectionProfileManager = InspectionProfileManager.getInstance()
+            val inspectionProfileImpl = inspectionProfileManager.currentProfile
+            val model = InspectionProfileModifiableModel(inspectionProfileImpl)
+            model.modifyProfile {
```
Comment (iromeo) [inline src/main/kotlin/com/jetbrains/snakecharm/inspections/quickfix/AddIgnoredElementQuickFix.kt:38]: In PyCharm `src/com/jetbrains/python/inspections/quickfix/AddIgnoredIdentifierQuickFix.java` uses shorter solution for this usecase (i.e. `modifyAndCommitProjectProfile()`), please refactor this

File: src/main/kotlin/com/jetbrains/snakecharm/inspections/quickfix/AddIgnoredElementQuickFix.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                    SmkUnrecognizedSectionInspection::class.java.simpleName,
+                    file
+                ) as SmkUnrecognizedSectionInspection
+                if (text !in inspection.ignoredItems) {
+                    inspection.ignoredItems.add(startElement.text)
+                }
+            }
+            model.commit()
+            assert(editor != null)
+            TemplateBuilderImpl(startElement).run(editor!!, false)
```
Comment (iromeo) [inline src/main/kotlin/com/jetbrains/snakecharm/inspections/quickfix/AddIgnoredElementQuickFix.kt:49]: Why do we need this? E.g. `src/com/jetbrains/python/inspections/quickfix/AddIgnoredIdentifierQuickFix.java` doesn't have such code

File: src/main/kotlin/com/jetbrains/snakecharm/inspections/quickfix/AddIgnoredElementQuickFix.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.profile.codeInspection.InspectionProfileManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.jetbrains.snakecharm.SnakemakeBundle
+import com.jetbrains.snakecharm.inspections.SmkUnrecognizedSectionInspection
+
+class AddIgnoredElementQuickFix(element: PsiElement) : LocalQuickFixAndIntentionActionOnPsiElement(element) {
+    private val elementName = element.text
```
Comment (iromeo) [inline src/main/kotlin/com/jetbrains/snakecharm/inspections/quickfix/AddIgnoredElementQuickFix.kt:15]: Do we need here `LocalQuickFixAndIntentionActionOnPsiElement(element)` ? Could we use just ` implements LocalQuickFix, LowPriorityAction` ?

Comment (iromeo) [reply src/main/kotlin/com/jetbrains/snakecharm/inspections/quickfix/AddIgnoredElementQuickFix.kt:15]: Is `elementName` is section name? If yes, please use user-friendly variable name naming

File: src/main/kotlin/com/jetbrains/snakecharm/inspections/quickfix/AddIgnoredElementQuickFix.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    override fun invoke(
+        project: Project,
+        file: PsiFile,
+        editor: Editor?,
+        startElement: PsiElement,
+        endElement: PsiElement
+    ) {
+        val virtualFile = startElement.containingFile.virtualFile
+        if (virtualFile != null) {
+            val text = startElement.text
```
Comment (iromeo) [inline src/main/kotlin/com/jetbrains/snakecharm/inspections/quickfix/AddIgnoredElementQuickFix.kt:34]: is here text is same as `private val elementName = element.text` field? if yes, please use `elementName`

**Review by iromeo — CHANGES_REQUESTED** [review]

**Comment by dakochik** [comment]
refactored AddIgnoredElementQuickFix.kt code:
1) start using appropriate methods
2) change list of inherited classes
3) makes code meets Kotlin code style conventions
4) removes useless checks

memorizing

File: src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkUnrecognizedSectionInspection.kt [hunk 5]
```diff
@@ -59,7 +59,9 @@ class SmkUnrecognizedSectionInspection : SnakemakeInspection() {
     }
 
     override fun createOptionsPanel(): JComponent? {
-        return ListEditForm(SnakemakeBundle.message("INSP.NAME.section.unrecognized.ignored"),
-            ignoredItems).contentPanel
+        return ListEditForm(
```
Comment (iromeo) [inline src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkUnrecognizedSectionInspection.kt:62]: Please reformat according to Kotlin code style conventions:
> Prefer using an expression body for functions with the body consisting of a single expression.

See https://kotlinlang.org/docs/coding-conventions.html#functions

File: src/main/kotlin/com/jetbrains/snakecharm/inspections/quickfix/AddIgnoredElementQuickFix.kt [hunk 6]
```diff
@@ -1,52 +1,30 @@
...
-import com.intellij.profile.codeInspection.InspectionProfileManager
 import com.intellij.psi.PsiElement
-import com.intellij.psi.PsiFile
 import com.jetbrains.snakecharm.SnakemakeBundle
 import com.jetbrains.snakecharm.inspections.SmkUnrecognizedSectionInspection
 
-class AddIgnoredElementQuickFix(element: PsiElement) : LocalQuickFixAndIntentionActionOnPsiElement(element) {
-    private val elementName = element.text
+class AddIgnoredElementQuickFix(section: PsiElement) : LocalQuickFix, LowPriorityAction {
+    private val sectionName = section.text
```
Comment (iromeo) [inline src/main/kotlin/com/jetbrains/snakecharm/inspections/quickfix/AddIgnoredElementQuickFix.kt:12]: `section: PsiElement` used only once for private field initialization, let's simply constructor to `class AddIgnoredElementQuickFix(private val sectionName: String)`

**Review by iromeo — CHANGES_REQUESTED** [review]
Some minor code improvements still required

**Comment by dakochik** [comment]
refactored: 
1) simplifies AddIgnoredElementQuickFix.kt constructor
2) makes SmkUnrecognizedSectionInspection.kt code meets Kotlin code style conventions

File: src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkUnrecognizedSectionInspection.kt [hunk 7]
```diff
@@ -31,7 +31,7 @@ class SmkUnrecognizedSectionInspection : SnakemakeInspection() {
                     sectionNamePsi,
                     SnakemakeBundle.message("INSP.NAME.section.unrecognized.message", sectionKeyword),
                     ProblemHighlightType.WEAK_WARNING,
-                    null, AddIgnoredElementQuickFix(sectionNamePsi)
+                    null, AddIgnoredElementQuickFix(sectionNamePsi.text)
```
Comment (iromeo) [inline src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkUnrecognizedSectionInspection.kt:34]: The section name is already stored in `sectionKeyword` variable, please use it instead

File: src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkUnrecognizedSectionInspection.kt [hunk 8]
```diff
@@ -46,7 +46,7 @@ class SmkUnrecognizedSectionInspection : SnakemakeInspection() {
...
```
Comment (iromeo) [inline src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkUnrecognizedSectionInspection.kt:49]: The section name is already stored in `sectionKeyword` variable, please use it instead

**Review by iromeo — CHANGES_REQUESTED** [review]
Minor code change still required

**Comment by dakochik** [comment]
refactored: makes function call more readable

**Pull request merged**